### ISSUE TITLE
Support newable in interfaces

### ISSUE THREAD

**Issue by shantanuraj** [palantir/tslint#3326]
### Bug Report

- __TSLint version__: 5.7.0
- __TypeScript version__: 2.5.2
- __Running TSLint via__: CLI

#### TypeScript code being linted

```ts
// Sample interface
interface Foo {
  concat: (suffix: string) => string;
}

// Sample interface using newable
interface Bar {
  new (prefix: string) => Foo;
}
```

with `tslint.json` configuration:

```json
{
  "defaultSeverity": "error",
  "extends": [
    "tslint:recommended",
    "tslint-config-prettier"
  ],
  "rules": {
    "no-unused-expression": [true, "allow-fast-null-checks"],
    "object-literal-key-quotes": [true, "as-needed"],
    "object-literal-sort-keys":  false,
    "quotemark": [true, "single", "jsx-double"],
    "sort-keys": ["error", "asc", {"caseSensitive": true, "natural": true}]
  }
}
```

#### Actual behavior

```
ERROR: /path-to-file[16, 3]: Interfaces cannot be constructed, only classes. Did you mean `declare class`?
```

#### Expected behavior

No error, should lint successfully. 

**Update:** Added correct error. Previously I had pasted the wrong error here, my bad.

**Comment by ajafff** [comment]
I cannot reproduce this locally. From the error message I can deduce that it comes from the `member-ordering` rule. It may be a duplicate of #3251

**Comment by shantanuraj** [comment]
@ajafff yeah I think this might not be a tslint issue after all..
After running the scripts manually the seem to work.

But as a pre-commit hook using husky tslint fails
Here's the screenshot for what its worth

![screen shot 2017-10-17 at 1 17 54 am](https://user-images.githubusercontent.com/1716853/31632006-1ecfca68-b2d9-11e7-904f-b474a32af09a.png)

**Comment by ajafff** [comment]
Just disable `no-misused-new` in your config.

We probably won't change that behavior, because that's what this rule does.

**Comment by ajafff** [comment]
I looked at the code to verify the rule works as expected. That's not the original code snippet you provided above. The rule only complains if the construct signature returns the same type as the interface:

```ts
interface Bar {
  new (prefix: string) => Bar; // this is not allowed. if the return type is something else, it's totally fine
}
```

**Comment by shantanuraj** [comment]
@ajafff got it makes sense, thanks!
I think it's safe to close this one.
